account_betas.go

```diff
@@ -102,13 +80,11 @@ func (c *Client) JoinBetaProgram(ctx context.Context, opts AccountBetaProgramCre
 	if err != nil {
 		return nil, err
 	}
-
 	e := "account/betas"
-	req := c.R(ctx).SetResult(&AccountBetaProgram{}).SetBody(string(body))
-	r, err := coupleAPIErrors(req.Post(e))
+	response, err := doPOSTRequest[AccountBetaProgram](ctx, c, e, body)
```

Comment:
The new `doPOSTRequest` and `doPUTRequest` helpers accept instances of structs and handle all of the serialization behind the scenes, so we should be able to pass in `opts` and remove the JSON marshaling logic above 🙂 

```suggestion
	 response, err := doPOSTRequest[AccountBetaProgram](ctx, c, e, opts)
```